src/reproducibility.py

```
@@ -0,0 +1,421 @@
+    optimizer = torch.optim.Adam(model.parameters(), lr=TRAIN_CONFIG["lr"])
+    logger = TelemetryLogger(filepath=log_file)
+
+    if checkpoint_path_to_load:
+        checkpoint = torch.load(checkpoint_path_to_load, weights_only=False)
+        model.load_state_dict(checkpoint['model'])
+
+        # verifying cryptographic seal
+        if 'checkpoint_hash' in checkpoint:
+            loaded_hash = logger.hash_model(model)
+            if loaded_hash != checkpoint['checkpoint_hash']:
+                print(f"\n  FATAL ALERT! : Cryptographic seal broken! Checkpoint file was tampered with.")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)